### ISSUE TITLE
Check all command line options

### ISSUE THREAD

**Issue by rprouse** [nunit/dotnet-test-nunit#4]
Make sure that all command line options listed are supported and make sure none that should be supported are missed. Need to distinguish between framework options and engine options that may not be supported.

**Comment by roji** [comment]
+1, if I understand correctly this is the reason I can't pass --teamcity when using `dotnet test`.

**Comment by CharliePoole** [comment]
Because the teamcity extension is being split away from NUnit, it might make sense that this runner not support it at this time. Right now, we support this for the engine and nunitlite using entirely different code - so different that there's an issue about potentially resolving the differences. Creating a third implementation is possible, of course, but only adds to the messiness.

My understanding is that you can set up teamcity so it will run tests without getting the special messages provided by our teamcity option. This may require producing an XML result in nunit2 format, however, which may be even messier than a third implementation.

Just so you're all aware of the issues!

**Comment by rprouse** [comment]
@roji, TeamCity is an engine extension and this runner does not use the engine, so the `--teamcity` option is not implemented. I don't think we will be adding direct TeamCity integration into this runner either. To be honest, I find it a bit odd that TeamCity requires (prefers?) custom integration code in test runners rather than just capturing Console.Out and parsing the test results XML file like every other CI system.

**Comment by roji** [comment]
I don't know much about TeamCity and its integration with test runners. I can imagine that parsing an XML means TeamCity would have to wait until the end of the test run before processing any results, whereas the special stdout integration allows it to understand what's going on in realtime, reflecting that in the UI etc. (although I may be wrong).

I also know very little about internal NUnit architecture, was just expecting this runner to support it just like nunitlite and nunit-console do...

**Comment by CharliePoole** [comment]
That's right. They want something that lets them monitor execution as it proceeds. There are other ways to do that, however, than generating special messages to the specification of one particular vendor. NUnit issues events whenever a test or suite starts or finishes. Everything there is to know about test progress is available. What the teamcity extension does is convert those events into a proprietary TC format and send it to stdout.

Going along with this approach is simply a mistake we made some time ago. We haven't accomodated any other vendor in that way and they all manage to interface with us because it's in their economic interest. Probably we can't roll back the decisions we made in the past, but we can avoid extending them.

Currently, the code that does all this is mostly maintained by a guy who works for JetBrains, which at least puts things in the right perspective by my lights. :-) The code used by the console runner is now in an extension assembly and will be moved to a separate project repo rather sooner than later. With a few tweaks, I think we can make that extension useable by nunitlite as well, without the overhead of the engine and it's extensibility framework. At that point, it should be useable by any runner that wants to go to the trouble. OTOH, maybe we'll be running dotnet tests through the console runner before that happens. :-)
